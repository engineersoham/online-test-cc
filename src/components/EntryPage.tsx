import { Grid, Typography, List, ListItem, ListItemButton, ListItemIcon, 
    Box, Select, InputLabel, FormControl, MenuItem, SelectChangeEvent, Button,  Table, TableRow, TableCell } from '@mui/material'
 import React from 'react'
 import { useNavigate } from 'react-router-dom';
 import StarIcon from '@mui/icons-material/Star';
 
 
 type Props = {
   setLanguage:any
 }
 
 const EntryPage: React.FC<Props> = ({ ...props }) => {
   const [language, setLanguage] = React.useState('');
   const navigate = useNavigate()
 
   const handleChange = (event: SelectChangeEvent) => {
     setLanguage(event.target.value);
   };
   const submit = () => {
       navigate('/quiz')
     props.setLanguage(language)
   }
   return (
 
     <Grid container sx={{ ml: 10 }}>
       
       <List
         sx={{ width: '100%', bgcolor: 'background.paper', mt: 10 }}
         aria-label="contacts"
       >
         <ListItem disablePadding>
           <ListItemButton>
             <ListItemIcon>
               <StarIcon />
             </ListItemIcon>
             <Typography>Select the language. then start the test.</Typography>
           </ListItemButton>
         </ListItem>
 
 
 
         <ListItem disablePadding>
           <ListItemButton>
             <ListItemIcon>
               <StarIcon />
             </ListItemIcon>
             <Typography>
               There are five Questions, One of each question will be of type: multiple choice, true/false, 
               fill-in the blanks, match the following and multi select.
             </Typography>
               
           </ListItemButton>
         </ListItem>
 
         <ListItem disablePadding>
           <ListItemButton>
             <ListItemIcon>
               <StarIcon />
             </ListItemIcon>
             <Typography>You can switch between questions.</Typography>
           </ListItemButton>
         </ListItem>
 
         <ListItem disablePadding>
           <ListItemButton>
             <ListItemIcon>
               <StarIcon />
             </ListItemIcon>
             <Typography>You can select the perticular question by clicking on the perticular question showing at left side of the page.</Typography>
           </ListItemButton>
         </ListItem>
 
         <ListItem disablePadding>
           <ListItemButton>
             <ListItemIcon>
               <StarIcon />
             </ListItemIcon>
             <Typography>Answered questions will be shown in green color and unanswered in grey color.</Typography>
           </ListItemButton>
         </ListItem>
 
         <ListItem disablePadding>
           <ListItemButton>
             <ListItemIcon>
               <StarIcon />
             </ListItemIcon>
             <Typography>Make sure that you save the current answer before moving to next question.</Typography>
           </ListItemButton>
         </ListItem>
 
       </List>
 
       <Box sx={{ mt: 15, ml: 80 }}>
         <FormControl >
           <InputLabel >Selact Language</InputLabel>
           <Table>
             <TableRow>
               <TableCell>
                 <Select
                   sx={{ width: 300, m: 2 }}
                   value={language}
                   onChange={handleChange}
                   defaultValue='Select'
                 >
                   <MenuItem value='English'>English</MenuItem>
                   <MenuItem value='Hindi'>Hindi</MenuItem>
                   <MenuItem value='Kannada'>Kannada</MenuItem>
                   
                    </Select>
               </TableCell>
               <TableCell>
                 <Button title='startTest' sx={{ width: 200, ml: 5 }} style={{ backgroundColor: 'rgb(18, 114, 209)', color: 'white' }} variant='outlined' onClick={submit}>Start Test</Button>
               </TableCell>
             </TableRow>
           </Table>
         </FormControl>
       </Box>
 
     </Grid>
   )
 }
 
 export default EntryPage