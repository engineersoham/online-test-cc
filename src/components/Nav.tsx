import * as React from 'react';
import { AppBar, Box, Toolbar, Typography} from '@mui/material';
import { Link } from 'react-router-dom'

const Nav = () => {
    return (
        <Box sx={{ flexGrow: 1 }}>
            <AppBar position="sticky">
                <Toolbar>
                    <Typography variant="h6" component="div" sx={{ flexGrow: 1, ml:80 }}>
                        Online Mocktest
                    </Typography>
                    <Typography variant="h6" component="div" sx={{ flexGrow: 0, mr:8 }}>
                        <Link style={{ textDecoration: 'none', color:'white' }} to='/login'>Login</Link>
                    </Typography>
                    <Typography variant="h6" component="div" sx={{ flexGrow: 0 }}>
                        <Link style={{ textDecoration: 'none', color:'white' }} to='/signup'>Sign Up</Link>
                    </Typography>


                </Toolbar>
            </AppBar>
        </Box>
    );
}
export default Nav