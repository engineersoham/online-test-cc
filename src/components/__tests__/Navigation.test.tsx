import { Pagination } from '@mui/material';
import {screen, render, fireEvent, getByRole} from '@testing-library/react'
import { Provider } from 'react-redux';
import {BrowserRouter, Link} from 'react-router-dom';
import { store } from '../../Store';
import EntryPage from '../EntryPage';
import Nav from '../Nav';
import SignUp from '../SignUp';


test('get snapshot of navbar', ()=>{
    const tree = render(
        <BrowserRouter><Nav></Nav></BrowserRouter>
    )
    expect(tree).toMatchSnapshot()
})

describe('test Signup button', ()=>{
    it('sign Up button test', ()=>{
        render(
            <BrowserRouter><Nav></Nav></BrowserRouter>
        )
        const text = screen.getByText('Sign Up')
        expect(text).toBeInTheDocument()
    });

    test('signup page test',()=>{
        render(
            <BrowserRouter><SignUp/></BrowserRouter>
        )
        expect(screen.getAllByRole('textbox').length).toBe(3)
        expect(screen.getAllByRole('radio').length).toBe(3)
        expect(screen.getByRole('button')).toBeInTheDocument()
    });

    test('button function',()=>{
        const mockfun = jest.fn();
        render(
            <BrowserRouter><EntryPage setLanguage={mockfun}/></BrowserRouter>
        );
        
        fireEvent.click(screen.getByText('Start Test'));
        expect(mockfun).toHaveBeenCalled();
    })

    test('test for enter page',()=>{
        render(
            <BrowserRouter><EntryPage setLanguage={()=>{}}/></BrowserRouter>
        );
        expect(screen.getByText('Start Test')).toBeInTheDocument()
            screen.debug()
    })

})