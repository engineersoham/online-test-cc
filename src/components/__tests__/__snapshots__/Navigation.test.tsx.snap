// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get snapshot of navbar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-i9gxme"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-7e0s21-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h6 css-1cx6i83-MuiTypography-root"
            >
              Online Mocktest
            </div>
            <div
              class="MuiTypography-root MuiTypography-h6 css-2ofoyk-MuiTypography-root"
            >
              <a
                href="/login"
                style="text-decoration: none; color: white;"
              >
                Login
              </a>
            </div>
            <div
              class="MuiTypography-root MuiTypography-h6 css-7fum8j-MuiTypography-root"
            >
              <a
                href="/signup"
                style="text-decoration: none; color: white;"
              >
                Sign Up
              </a>
            </div>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-i9gxme"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-7e0s21-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 css-1cx6i83-MuiTypography-root"
          >
            Online Mocktest
          </div>
          <div
            class="MuiTypography-root MuiTypography-h6 css-2ofoyk-MuiTypography-root"
          >
            <a
              href="/login"
              style="text-decoration: none; color: white;"
            >
              Login
            </a>
          </div>
          <div
            class="MuiTypography-root MuiTypography-h6 css-7fum8j-MuiTypography-root"
          >
            <a
              href="/signup"
              style="text-decoration: none; color: white;"
            >
              Sign Up
            </a>
          </div>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
